### PR TITLE
Build fixes

### PR DESCRIPTION
**Pull request by proller**
Fixed build on Mac OS X.
No changes in behaviour.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
> No description provided.

What build fixes?

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,15 @@
 project(ClickHouse)
 cmake_minimum_required(VERSION 3.3)
-cmake_policy(SET CMP0023 NEW)
+
+foreach(policy
+    CMP0023
```
Comment (alexey-milovidov) [inline CMakeLists.txt]: What does it mean?

**Comment by alexey-milovidov** [comment]
I have provided description for you.

**Comment by filimonov** [comment]
For googling. Fixes following:

1. Can't find ICU
```-- The following ICU libraries were not found:
--   i18n (required)
--   uc (required)
--   data (required)
-- Failed to find all ICU components (missing: ICU_INCLUDE_DIR ICU_LIBRARY _ICU_REQUIRED_LIBS_FOUND)
```
2. After adding ICU_ROOT:
```
CMake Warning (dev) at cmake/find_icu.cmake:8 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable ICU_ROOT is set to:

    /usr/local/opt/icu4c

  For compatibility, CMake is ignoring the variable.
Call Stack (most recent call first):
  CMakeLists.txt:288 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found the following ICU libraries:
--   i18n (required)
--   uc (required)
--   data (required)
-- Found ICU: /usr/local/opt/icu4c/include (found version "64.2")``` 

3. test for AIO
```
[2010/3305] Building CXX object dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o
FAILED: dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o 
/usr/local/bin/g++-8  -DBOOST_SYSTEM_NO_DEPRECATED -DGTEST_HAS_POSIX_RE=0 -DPOCO_STATIC -DPOCO_UNBUNDLED_ZLIB -DUNALIGNED_OK -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -I../contrib/googletest/googletest/include -I../dbms/src -Idbms/src -I../libs/libcommon/include -Ilibs/libcommon/include -I../contrib/zlib-ng -Icontrib/zlib-ng -I../contrib/cityhash102/include -I../contrib/croaring -isystem ../contrib/googletest/include -isystem ../contrib/pdqsort -isystem ../contrib/libpcg-random/include -isystem ../contrib/double-conversion -isystem ../contrib/lfalloc/src -isystem ../contrib/re2 -isystem ../contrib/poco/Foundation/include -isystem ../contrib/boost -isystem ../contrib/poco/Net/include -isystem ../contrib/poco/Util/include -isystem ../contrib/poco/XML/include -isystem ../contrib/poco/JSON/include -isystem contrib/re2_st -fdiagnostics-color=always  -msse4.1 -msse4.2 -mpopcnt  -fno-omit-frame-pointer  -Wall  -Wnon-virtual-dtor -Wextra -Werror -O2 -g -DNDEBUG -O3    -fno-tree-loop-distribute-patterns -std=c++17 -MD -MT dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o -MF dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o.d -o dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o -c ../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp: In member function 'virtual void ReadBufferAIOTest_TestReadAfterAIO_Test::TestBody()':
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp:42:21: error: 'DB' is not a namespace-name
     using namespace DB;
                     ^~
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp:42:23: error: expected namespace-name before ';' token
     using namespace DB;
                       ^
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp:46:5: error: 'ReadBufferAIO' was not declared in this scope
     ReadBufferAIO testbuf(file_path);
     ^~~~~~~~~~~~~
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp:51:25: error: 'testbuf' was not declared in this scope
     size_t total_read = testbuf.read(newdata.data(), newdata.length());
                         ^~~~~~~
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp:51:25: note: suggested alternative: 'setbuf'
     size_t total_read = testbuf.read(newdata.data(), newdata.length());
                         ^~~~~~~
                         setbuf
ninja: build stopped: subcommand failed
```

4. 
```
[2010/3305] Building CXX object dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o
FAILED: dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o 
/usr/local/bin/g++-8  -DBOOST_SYSTEM_NO_DEPRECATED -DGTEST_HAS_POSIX_RE=0 -DPOCO_STATIC -DPOCO_UNBUNDLED_ZLIB -DUNALIGNED_OK -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -I../contrib/googletest/googletest/include -I../dbms/src -Idbms/src -I../libs/libcommon/include -Ilibs/libcommon/include -I../contrib/zlib-ng -Icontrib/zlib-ng -I../contrib/cityhash102/include -I../contrib/croaring -isystem ../contrib/googletest/include -isystem ../contrib/pdqsort -isystem ../contrib/libpcg-random/include -isystem ../contrib/double-conversion -isystem ../contrib/lfalloc/src -isystem ../contrib/re2 -isystem ../contrib/poco/Foundation/include -isystem ../contrib/boost -isystem ../contrib/poco/Net/include -isystem ../contrib/poco/Util/include -isystem ../contrib/poco/XML/include -isystem ../contrib/poco/JSON/include -isystem contrib/re2_st -fdiagnostics-color=always  -msse4.1 -msse4.2 -mpopcnt  -fno-omit-frame-pointer  -Wall  -Wnon-virtual-dtor -Wextra -Werror -O2 -g -DNDEBUG -O3    -fno-tree-loop-distribute-patterns -std=c++17 -MD -MT dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o -MF dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o.d -o dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o -c ../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp: In member function 'virtual void ReadBufferAIOTest_TestReadAfterAIO_Test::TestBody()':
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp:42:21: error: 'DB' is not a namespace-name
     using namespace DB;
                     ^~
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp:42:23: error: expected namespace-name before ';' token
     using namespace DB;
                       ^
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp:46:5: error: 'ReadBufferAIO' was not declared in this scope
     ReadBufferAIO testbuf(file_path);
     ^~~~~~~~~~~~~
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp:51:25: error: 'testbuf' was not declared in this scope
     size_t total_read = testbuf.read(newdata.data(), newdata.length());
                         ^~~~~~~
../dbms/src/IO/tests/gtest_aio_seek_back_after_eof.cpp:51:25: note: suggested alternative: 'setbuf'
     size_t total_read = testbuf.read(newdata.data(), newdata.length());
                         ^~~~~~~
                         setbuf
ninja: build stopped: subcommand failed```

5. 
```
[106/1019] Building CXX object dbms/CMakeFiles/clickhouse_common_io.dir/src/Common/LFAllocator.cpp.o
FAILED: dbms/CMakeFiles/clickhouse_common_io.dir/src/Common/LFAllocator.cpp.o 
/usr/local/bin/g++-8  -DBOOST_SYSTEM_NO_DEPRECATED -DPOCO_STATIC -DPOCO_UNBUNDLED_ZLIB -DUNALIGNED_OK -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -I../libs/libcommon/include -Ilibs/libcommon/include -I../dbms/src -Idbms/src -I../contrib/zlib-ng -Icontrib/zlib-ng -I../contrib/cityhash102/include -I../libs/libwidechar_width -I../contrib/croaring -isystem ../contrib/double-conversion -isystem ../contrib/brotli/c/include -isystem ../contrib/lfalloc/src -isystem ../contrib/re2 -isystem ../contrib/poco/Data/include -isystem ../contrib/libpcg-random/include -isystem ../contrib/poco/Foundation/include -isystem ../contrib/boost -isystem ../contrib/poco/Net/include -isystem ../contrib/poco/Util/include -isystem ../contrib/poco/XML/include -isystem ../contrib/poco/JSON/include -isystem contrib/re2_st -isystem ../contrib/libcpuid/include -isystem ../contrib/poco/NetSSL_OpenSSL/include -isystem ../contrib/poco/Crypto/include -isystem ../contrib/ssl/include -fdiagnostics-color=always  -msse4.1 -msse4.2 -mpopcnt  -fno-omit-frame-pointer  -Wall  -Wnon-virtual-dtor -Wextra -Werror -O2 -g -DNDEBUG -O3    -fno-tree-loop-distribute-patterns -std=c++17 -MD -MT dbms/CMakeFiles/clickhouse_common_io.dir/src/Common/LFAllocator.cpp.o -MF dbms/CMakeFiles/clickhouse_common_io.dir/src/Common/LFAllocator.cpp.o.d -o dbms/CMakeFiles/clickhouse_common_io.dir/src/Common/LFAllocator.cpp.o -c ../dbms/src/Common/LFAllocator.cpp
../dbms/src/Common/LFAllocator.cpp: In static member function 'static void* DB::LFAllocator::alloc(size_t, size_t)':
../dbms/src/Common/LFAllocator.cpp:15:16: error: 'LFAlloc' was not declared in this scope
         return LFAlloc(size);
                ^~~~~~~
../dbms/src/Common/LFAllocator.cpp:15:16: note: suggested alternative: 'alloc'
         return LFAlloc(size);
                ^~~~~~~
                alloc
../dbms/src/Common/LFAllocator.cpp:19:19: error: 'LFPosixMemalign' was not declared in this scope
         int res = LFPosixMemalign(&ptr, alignment, size);
                   ^~~~~~~~~~~~~~~
../dbms/src/Common/LFAllocator.cpp:19:19: note: suggested alternative: 'posix_memalign'
         int res = LFPosixMemalign(&ptr, alignment, size);
                   ^~~~~~~~~~~~~~~
                   posix_memalign
../dbms/src/Common/LFAllocator.cpp: In static member function 'static void DB::LFAllocator::free(void*, size_t)':
../dbms/src/Common/LFAllocator.cpp:26:5: error: 'LFFree' was not declared in this scope
     LFFree(buf);
     ^~~~~~
../dbms/src/Common/LFAllocator.cpp:26:5: note: suggested alternative: 'free'
     LFFree(buf);
     ^~~~~~
     free
../dbms/src/Common/LFAllocator.cpp: In static member function 'static void* DB::LFAllocator::realloc(void*, size_t, size_t, size_t)':
../dbms/src/Common/LFAllocator.cpp:38:9: error: 'LFFree' was not declared in this scope
         LFFree(old_ptr);
         ^~~~~~
../dbms/src/Common/LFAllocator.cpp:38:9: note: suggested alternative: 'free'
         LFFree(old_ptr);
         ^~~~~~
         free
../dbms/src/Common/LFAllocator.cpp:45:23: error: 'LFGetSize' was not declared in this scope
     size_t old_size = LFGetSize(old_ptr);
                       ^~~~~~~~~
../dbms/src/Common/LFAllocator.cpp:47:5: error: 'LFFree' was not declared in this scope
     LFFree(old_ptr);
     ^~~~~~
../dbms/src/Common/LFAllocator.cpp:47:5: note: suggested alternative: 'free'
     LFFree(old_ptr);
     ^~~~~~
     free
ninja: build stopped: subcommand failed.
```

**Comment by filimonov** [comment]
Currenly Mac build with default settings fails on linking of ICU:

```[2/80] Linking CXX executable dbms/unit_tests_dbms
FAILED: dbms/unit_tests_dbms
: && /usr/local/bin/g++-8  -fdiagnostics-color=always  -msse4.1 -msse4.2 -mpopcnt  -fno-omit-frame-pointer  -Wall  -Wnon-virtual-dtor -Wextra -Werror -O2 -g -DNDEBUG -O3 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  dbms/CMakeFiles/unit_tests_dbms.dir/src/Common/ZooKeeper/tests/gtest_zkutil_test_multi_exception.cpp.o dbms/CMakeFiles/unit_tests_dbms.dir/src/Common/tests/gtest_rw_lock.cpp.o dbms/CMakeFiles/unit_tests_dbms.dir/src/Common/tests/gtest_unescapeForFileName.cpp.o dbms/CMakeFiles/unit_tests_dbms.dir/src/DataStreams/tests/gtest_blocks_size_merging_streams.cpp.o dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_aio_seek_back_after_eof.cpp.o dbms/CMakeFiles/unit_tests_dbms.dir/src/IO/tests/gtest_cascade_and_memory_write_buffer.cpp.o dbms/CMakeFiles/unit_tests_dbms.dir/src/Storages/tests/gtest_aux_funcs_for_adaptive_granularity.cpp.o dbms/CMakeFiles/unit_tests_dbms.dir/src/Storages/tests/gtest_row_source_bits_test.cpp.o  -o dbms/unit_tests_dbms  contrib/googletest/libgtest_main.a contrib/googletest/libgtest.a dbms/libdbms.a dbms/src/Common/ZooKeeper/libclickhouse_common_zookeeper.a dbms/src/Compression/libclickhouse_compression.a dbms/src/Parsers/libclickhouse_parsers.a dbms/src/Common/Config/libclickhouse_common_config.a dbms/src/Common/ZooKeeper/libclickhouse_common_zookeeper.a dbms/src/Dictionaries/Embedded/libclickhouse_dictionaries_embedded.a dbms/libclickhouse_common_io.a libs/libcommon/libcommon.a contrib/cctz-cmake/libcctz.a contrib/boost-cmake/libboost_filesystem_internal.a libs/libmemcpy/libmemcpy.a dbms/src/Common/StringUtils/libstring_utils.a contrib/cityhash102/libcityhash.a libs/libwidechar_width/libwidechar_width.a contrib/re2/libre2.a contrib/re2_st/libre2_st.a libs/libcommon/libapple_rt.a contrib/croaring/libroaring.a contrib/libcpuid/libcpuid.a contrib/brotli-cmake/libbrotli.a contrib/libbtrie/libbtrie.a contrib/boost-cmake/libboost_program_options_internal.a contrib/boost-cmake/libboost_system_internal.a contrib/poco/MongoDB/libPocoMongoDB.a contrib/poco/NetSSL_OpenSSL/libPocoNetSSL.a contrib/poco/Util/libPocoUtil.a contrib/poco/XML/libPocoXML.a contrib/poco/JSON/libPocoJSON.a contrib/poco/Net/libPocoNet.a contrib/poco/Crypto/libPocoCrypto.a contrib/ssl/ssl/libssl.a contrib/ssl/crypto/libcrypto.a contrib/poco/Foundation/libPocoFoundation.a -ldl /usr/local/opt/icu4c/lib/libicui18n.a /usr/local/opt/icu4c/lib/libicuuc.a /usr/local/opt/icu4c/lib/libicudata.a contrib/capnproto/c++/src/capnp/libcapnpc.a contrib/capnproto/c++/src/capnp/libcapnp.a contrib/capnproto/c++/src/kj/libkj.a -lpthread contrib/arrow-cmake/libparquet_static.a contrib/arrow-cmake/libarrow_static.a contrib/zstd-cmake/libzstd.a contrib/lz4-cmake/liblz4.a contrib/double-conversion/libdouble-conversion.a contrib/zlib-ng/libz.a contrib/snappy/libsnappy.a contrib/arrow-cmake/libthrift_static.a contrib/boost-cmake/libboost_regex_internal.a contrib/protobuf/cmake/libprotobuf.a && :
Undefined symbols for architecture x86_64:
  "std::__1::__shared_weak_count::__get_deleter(std::type_info const&) const", referenced from:
      vtable for std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> > in libicuuc.a(umutex.ao)
  "std::__1::condition_variable::notify_all()", referenced from:
      _umtx_condBroadcast_64 in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPostInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
  "std::__1::condition_variable::notify_one()", referenced from:
      _umtx_condSignal_64 in libicuuc.a(umutex.ao)
  "std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&)", referenced from:
      void std::__1::condition_variable_any::wait<std::__1::mutex>(std::__1::mutex&) in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPreInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
  "std::__1::condition_variable::~condition_variable()", referenced from:
      icu_64::UConditionVar::UConditionVar() in libicuuc.a(umutex.ao)
      icu_64::UConditionVar::~UConditionVar() in libicuuc.a(umutex.ao)
      icu_64::initCondition()       in libicuuc.a(umutex.ao)
  "std::__1::__shared_weak_count::__add_shared()", referenced from:
      void std::__1::condition_variable_any::wait<std::__1::mutex>(std::__1::mutex&) in libicuuc.a(umutex.ao)
  "std::__1::__shared_weak_count::__release_shared()", referenced from:
      icu_64::UConditionVar::~UConditionVar() in libicuuc.a(umutex.ao)
      void std::__1::condition_variable_any::wait<std::__1::mutex>(std::__1::mutex&) in libicuuc.a(umutex.ao)
  "std::__1::__shared_weak_count::~__shared_weak_count()", referenced from:
      std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> >::~__shared_ptr_emplace() in libicuuc.a(umutex.ao)
      std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> >::~__shared_ptr_emplace() in libicuuc.a(umutex.ao)
  "std::__1::__throw_system_error(int, char const*)", referenced from:
      std::__1::unique_lock<std::__1::mutex>::unlock() in libicuuc.a(umutex.ao)
  "std::__1::mutex::lock()", referenced from:
      _umtx_lock_64 in libicuuc.a(umutex.ao)
      void std::__1::condition_variable_any::wait<std::__1::mutex>(std::__1::mutex&) in libicuuc.a(umutex.ao)
      _umtx_condBroadcast_64 in libicuuc.a(umutex.ao)
      _umtx_condSignal_64 in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPreInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPostInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
  "std::__1::mutex::unlock()", referenced from:
      _umtx_unlock_64 in libicuuc.a(umutex.ao)
      void std::__1::condition_variable_any::wait<std::__1::mutex>(std::__1::mutex&) in libicuuc.a(umutex.ao)
      _umtx_condBroadcast_64 in libicuuc.a(umutex.ao)
      _umtx_condSignal_64 in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPreInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPostInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
      std::__1::unique_lock<std::__1::mutex>::unlock() in libicuuc.a(umutex.ao)
      ...
  "std::__1::mutex::~mutex()", referenced from:
      icu_64::UMutex::~UMutex() in libicuuc.a(unifiedcache.ao)
      icu_64::initMutex()      in libicuuc.a(umutex.ao)
      icu_64::UMutex::~UMutex() in libicuuc.a(umutex.ao)
      std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> >::~__shared_ptr_emplace() in libicuuc.a(umutex.ao)
      std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> >::~__shared_ptr_emplace() in libicuuc.a(umutex.ao)
      std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> >::__on_zero_shared() in libicuuc.a(umutex.ao)
      icu_64::UMutex::~UMutex() in libicuuc.a(locid.ao)
      ...
  "typeinfo for std::__1::__shared_weak_count", referenced from:
      typeinfo for std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> > in libicuuc.a(umutex.ao)
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
[3/80] Linking CXX executable utils/convert-month-partitioned-parts/convert-month-partitioned-parts
FAILED: utils/convert-month-partitioned-parts/convert-month-partitioned-parts
: && /usr/local/bin/g++-8  -fdiagnostics-color=always  -msse4.1 -msse4.2 -mpopcnt  -fno-omit-frame-pointer  -Wall  -Wnon-virtual-dtor -Werror -O2 -g -DNDEBUG -O3 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  utils/convert-month-partitioned-parts/CMakeFiles/convert-month-partitioned-parts.dir/main.cpp.o  -o utils/convert-month-partitioned-parts/convert-month-partitioned-parts  dbms/libdbms.a dbms/src/Parsers/libclickhouse_parsers.a contrib/boost-cmake/libboost_program_options_internal.a dbms/src/Compression/libclickhouse_compression.a dbms/src/Parsers/libclickhouse_parsers.a dbms/src/Common/Config/libclickhouse_common_config.a dbms/src/Common/ZooKeeper/libclickhouse_common_zookeeper.a dbms/src/Dictionaries/Embedded/libclickhouse_dictionaries_embedded.a dbms/libclickhouse_common_io.a libs/libcommon/libcommon.a contrib/cityhash102
/libcityhash.a contrib/cctz-cmake/libcctz.a contrib/boost-cmake/libboost_filesystem_internal.a libs/libmemcpy/libmemcpy.a dbms/src/Common/StringUtils/libstring_utils.a libs/libwidechar_width/libwidechar_width.a contrib/re2/libre2.a contrib/re2_st/libre2_st.a libs/libcommon/libapple_rt.a contrib/croaring/libroaring.a contrib/libcpuid/libcpuid.a contrib/brotli-cmake/libbrotli.a contrib/libbtrie/libbtrie.a contrib/boost-cmake/libboost_system_internal.a contrib/poco/MongoDB/libPocoMongoDB.a contrib/poco/NetSSL_OpenSSL/libPocoNetSSL.a contrib/poco/Util/libPocoUtil.a contrib/poco/XML/libPocoXML.a contrib/poco/JSON/libPocoJSON.a contrib/poco/Net/libPocoNet.a contrib/poco/Crypto/libPocoCrypto.a contrib/ssl/ssl/libssl.a contrib/ssl/crypto/libcrypto.a contrib/poco/Foundation/libPocoFoundation.a -ldl /usr/local/opt/icu4c/lib/libicui18n.a /usr/local/opt/icu4c/lib/libicuuc.a /usr/local/opt/icu4c/lib/libicudata.a contrib/capnproto/c++/src/capnp/libcapnpc.a contrib/capnproto/c++/src/capnp/libcapnp.a contrib/capnproto/c++/src/kj/libkj.a -lpthread contrib/arrow-cmake/libparquet_static.a contrib/arrow-cmake/libarrow_static.a contrib/zstd-cmake/libzstd.a contrib/lz4-cmake/liblz4.a contrib/double-conversion/libdouble-conversion.a contrib/zlib-ng/libz.a contrib/snappy/libsnappy.a contrib/arrow-cmake/libthrift_static.a contrib/boost-cmake/libboost_regex_internal.a contrib/protobuf/cmake/libprotobuf.a && :
Undefined symbols for architecture x86_64:
  "std::__1::__shared_weak_count::__get_deleter(std::type_info const&) const", referenced from:
      vtable for std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> > in libicuuc.a(umutex.ao)
  "std::__1::condition_variable::notify_all()", referenced from:
      _umtx_condBroadcast_64 in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPostInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
  "std::__1::condition_variable::notify_one()", referenced from:
      _umtx_condSignal_64 in libicuuc.a(umutex.ao)
  "std::__1::condition_variable::wait(std::__1::unique_lock<std::__1::mutex>&)", referenced from:
      void std::__1::condition_variable_any::wait<std::__1::mutex>(std::__1::mutex&) in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPreInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
  "std::__1::condition_variable::~condition_variable()", referenced from:
      icu_64::UConditionVar::UConditionVar() in libicuuc.a(umutex.ao)
      icu_64::UConditionVar::~UConditionVar() in libicuuc.a(umutex.ao)
      icu_64::initCondition()       in libicuuc.a(umutex.ao)
  "std::__1::__shared_weak_count::__add_shared()", referenced from:
      void std::__1::condition_variable_any::wait<std::__1::mutex>(std::__1::mutex&) in libicuuc.a(umutex.ao)
  "std::__1::__shared_weak_count::__release_shared()", referenced from:
      icu_64::UConditionVar::~UConditionVar() in libicuuc.a(umutex.ao)
      void std::__1::condition_variable_any::wait<std::__1::mutex>(std::__1::mutex&) in libicuuc.a(umutex.ao)
  "std::__1::__shared_weak_count::~__shared_weak_count()", referenced from:
      std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> >::~__shared_ptr_emplace() in libicuuc.a(umutex.ao)
      std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> >::~__shared_ptr_emplace() in libicuuc.a(umutex.ao)
  "std::__1::__throw_system_error(int, char const*)", referenced from:
      std::__1::unique_lock<std::__1::mutex>::unlock() in libicuuc.a(umutex.ao)
  "std::__1::mutex::lock()", referenced from:
      _umtx_lock_64 in libicuuc.a(umutex.ao)
      void std::__1::condition_variable_any::wait<std::__1::mutex>(std::__1::mutex&) in libicuuc.a(umutex.ao)
      _umtx_condBroadcast_64 in libicuuc.a(umutex.ao)
      _umtx_condSignal_64 in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPreInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPostInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
  "std::__1::mutex::unlock()", referenced from:
      _umtx_unlock_64 in libicuuc.a(umutex.ao)
      void std::__1::condition_
variable_any::wait<std::__1::mutex>(std::__1::mutex&) in libicuuc.a(umutex.ao)
      _umtx_condBroadcast_64 in libicuuc.a(umutex.ao)
      _umtx_condSignal_64 in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPreInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
      icu_64::umtx_initImplPostInit(icu_64::UInitOnce&)  in libicuuc.a(umutex.ao)
      std::__1::unique_lock<std::__1::mutex>::unlock() in libicuuc.a(umutex.ao)
      ...
  "std::__1::mutex::~mutex()", referenced from:
      icu_64::UMutex::~UMutex() in libicuuc.a(unifiedcache.ao)
      icu_64::initMutex()      in libicuuc.a(umutex.ao)
      icu_64::UMutex::~UMutex() in libicuuc.a(umutex.ao)
      std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> >::~__shared_ptr_emplace() in libicuuc.a(umutex.ao)
      std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> >::~__shared_ptr_emplace() in libicuuc.a(umutex.ao)
      std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> >::__on_zero_shared() in libicuuc.a(umutex.ao)
      icu_64::UMutex::~UMutex() in libicuuc.a(locid.ao)
      ...
  "typeinfo for std::__1::__shared_weak_count", referenced from:
      typeinfo for std::__1::__shared_ptr_emplace<std::__1::mutex, std::__1::allocator<std::__1::mutex> > in libicuuc.a(umutex.ao)
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
[4/80] Building CXX object dbms/programs/performance-test/CMakeFiles/clickhouse-performance-test-lib.dir/ReportBuilder.cpp.o
ninja: build stopped: subcommand failed.
```

Workaround is to disable ICU for cmake. 
```cmake .. -DCMAKE_CXX_COMPILER=g++-8 -DCMAKE_C_COMPILER=gcc-8 -DENABLE_ICU=OFF```

After that Mac build succeeds.

**Comment by andyyzh** [comment]
In my MacOS, I update to the latest master branch code.
If not add ‘ - DENABLE_ICU = OFF’ link failed, If add it then build passed.

Icu4c is installed, is it version problem?

$ brew search icu
==> Formulae
icu4c ✔

**Comment by filimonov** [comment]
Can you try to execute that before  compilation? 

```
brew install icu4c
brew link icu4c --force
```

**Pull request merged**